### PR TITLE
[PM-14826] Add UsePolicies check to GET endpoints

### PR DESCRIPTION
**Pull request by JimmyVo16**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-14826

## 📔 Objective

1. Add a UsePolicies check to `GET organizations/{{orgId}}/policies/master-password` and `GET organizations/{{orgId}}/policies/token`
4. Add unit tests to account for these edge cases.
5. Update a unit to account for the happy path.

## 📸 Screenshots

manually tested the  `GET organizations/{{orgId}}/policies/token` 

For a free Org (sad path).

image

For an enterprise Org (happy path).

image

For `GET organizations/{{orgId}}/policies/master-password` 

Sad path
image

Happy path
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: src/Api/AdminConsole/Controllers/PoliciesController.cs [hunk 1]
```diff
@@ -62,25 +63,18 @@ public PoliciesController(
         _orgUserInviteTokenDataFactory = orgUserInviteTokenDataFactory;
         _featureService = featureService;
         _organizationHasVerifiedDomainsQuery = organizationHasVerifiedDomainsQuery;
+        _organizationRepository = organizationRepository;
     }
 
     [HttpGet("{type}")]
     public async Task<PolicyDetailResponseModel> Get(Guid orgId, int type)
     {
-        if (!await _currentContext.ManagePolicies(orgId))
```
[the remark this answers is not in the document]
Comment (r-tome) [reply src/Api/AdminConsole/Controllers/PoliciesController.cs:70]: Reordering the private properties is fine, but please re-add all the curly brackets to keep the code consistent with the rest of the codebase.

File: test/Api.Test/Controllers/PoliciesControllerTests.cs [hunk 2]
```diff
@@ -194,11 +236,42 @@ public async Task Get_WhenUserCannotManagePolicies_ThrowsNotFoundException(
...
+        // Arrange
+        organization.UsePolicies = false;
+
+        var organizationRepository = sutProvider.GetDependency<IOrganizationRepository>();
+        organizationRepository.GetByIdAsync(orgId).Returns(organization);
+
+
+        // Act & Assert
+        await Assert.ThrowsAsync<NotFoundException>(() =>
+            sutProvider.Sut.GetByToken(orgId, email, token, organizationUserId));
```
[the remark this answers is not in the document]
Comment (r-tome) [reply test/Api.Test/Controllers/PoliciesControllerTests.cs:261]: Feel free to add more unit tests if you'd like! While it's not expected for work outside the ticket scope, I'm a big advocate for maximizing test coverage. I really appreciate seeing that effort, kudos!

**Review by r-tome — CHANGES_REQUESTED** [review]
Great work! Just the code format to fix and its good to go. I'll also review the new unit tests if you decide to add them in

**Review by r-tome — DISMISSED** [review]
Great work!

**Comment by JimmyVo16** [comment]
I did a sanity check by starting the Admin service. 

image
